### PR TITLE
Feat: Lesson17 (3. Create arrow buttons with disable switch function and the number pagination)

### PR DESCRIPTION
**Pull request by yuka830**
# [Lesson17](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#%E3%82%B9%E3%83%A9%E3%82%A4%E3%83%89%E3%82%B7%E3%83%A7%E3%83%BC%E4%BD%9C%E6%88%90)
```
仕様
1.画面遷移してから3秒後に解決されるPromiseが返すオブジェクトを元にimgタグを5つつくる。

2.それぞれは.z-indexで重ねた状態。矢印画像をクリックを押すとスライド画像が変わる

3.5枚中何枚目かを表示して、5/5の場合Nextの矢印はdisabledにする。1/5枚の時はBackボタンはdisabledにする
```
## [CodeSandbox](https://codesandbox.io/s/lesson17-create-arrow-buttons-with-disable-switch-function-and-the-number-pagination-6iz4k?file=/main.js)
Last commit: [043aaca](https://github.com/yuka830/js-lessons/pull/22/commits/043aacadbd08cfb4ff046f8997c18c4d776b4e23)

This time I implemented no.3 in the above specification lists.
I implemented disable switch function and number pagination using index and array.length of images.

### CODE REVIEW

File: lesson17/main.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+const slideshowWrap = document.getElementById("js-slideshow");
+const ul = document.getElementById("js-img-list");
+let imgArray = [];
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
```
Comment (csakyo) [inline lesson17/main.js]: I thought this function is very useful! I will use it as a reference.

File: lesson17/main.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  const pagination = createElementWithClassName("p", "number-pagination");
+  slideshowWrap.insertAdjacentElement("afterend", pagination);
+  getCurrentPageNum();
+};
+
+const getCurrentPageNum = () => {
+  const pagination = document.querySelector(".number-pagination");
+  pagination.textContent = `${getCurrentImgIndex() + 1}/${imgArray.length}`;
+};
+
```
Comment (csakyo) [inline lesson17/main.js:152]: I think it is good about the adding disable functionality and creating pagination!

**Review by csakyo — APPROVED** [review]

File: lesson17/main.js [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  const isShowImg = document.querySelector(".is-show");
+  const currentImgIndex = imgArray.indexOf(isShowImg);
+  return currentImgIndex;
+};
+
+const switchDisableForBtn = () => {
+  const nextBtn = document.getElementById("js-next-btn");
+  const backBtn = document.getElementById("js-back-btn");
+
+  if (getCurrentImgIndex() === 0) {
```
Comment (nyattsu72) [inline lesson17/main.js:129]: I've never seen a function available in an if statement!
I learned a lot. Thank you!

**Review by nyattsu72 — APPROVED** [review]

File: lesson17/main.js (hunk 1 above)
Comment (yuka830) [reply lesson17/main.js]: Thank you. I use it registering it in snippets :)

File: lesson17/main.js (hunk 2 above)
Comment (yuka830) [reply lesson17/main.js:152]: Thank you so much!

File: lesson17/main.js (hunk 3 above)
Comment (yuka830) [reply lesson17/main.js:129]: Thank you for reviewing!

**Pull request merged**